packages/migrate/src/utils/warnDatasourceDriverAdapter.ts

```diff
@@ -2,6 +2,9 @@ import { ErrorCapturingSqlDriverAdapterFactory } from '@prisma/driver-adapter-ut
 import { SchemaContext } from '@prisma/internals'
 
 const DEPRECATED_PROPERTIES = ['url', 'directUrl', 'shadowDatabaseUrl']
+// Due to the complexity of making the datasource.url truly optional across the codebase
+// this placeholder value is used in the PSL parser if no URL property is given.
+const NO_URL_PLACEHOLDER = '<invalid>'
```

Comment:
Shouldn't is be `'<N/A>'`? 

It is not "invalid", just not specified.

Reply:
Fair point - I just went with what prisma-engines sets it to.I'll merge this in anyways for now and see down the line how to improve this also in line with our test setup.